### PR TITLE
[Ready]Usage of default member initializer

### PR DESCRIPTION
**Pull request by ShFil119**
Pros:
- Cleanup: actually some classes as initialized in headers, some in constructor in cpp. 
- Should make headers more readable.
- Reduces time of deps to rebuild.
- Don't need to remember to set values when we are adding more ctors.
- compiler checks if is value valid (see https://github.com/rwengine/openrw/commit/625ebdcb90f4f7e720793458ba056a1704d63218)

### CODE REVIEW

**Comment by husho** [comment]
Needs rebase

**Comment by ShFil119** [comment]
Rebased.

File: rwengine/src/engine/GameData.hpp [hunk 1]
```diff
@@ -60,7 +60,7 @@ class GameData {
     GameData(Logger* log, const rwfs::path& path);
     ~GameData();
```
Comment (Lihis) [inline rwengine/src/engine/GameData.hpp]: This should be `default`.

Comment (ShFil119) [reply rwengine/src/engine/GameData.hpp]: Done. ;)

File: rwengine/src/render/WaterRenderer.hpp [hunk 2]
```diff
@@ -41,19 +41,19 @@ class WaterRenderer {
     void render(GameRenderer* renderer, GameWorld* world);
 
 private:
-    std::unique_ptr<Renderer::ShaderProgram> waterProg;
-    std::unique_ptr<Renderer::ShaderProgram> maskProg;
+    std::unique_ptr<Renderer::ShaderProgram> waterProg = nullptr;
+    std::unique_ptr<Renderer::ShaderProgram> maskProg = nullptr;
```
Comment (danhedron) [inline rwengine/src/render/WaterRenderer.hpp]: nit: This is already the default value for ``unique_ptr``

**Pull request merged**